### ISSUE TITLE
Upgrading pyinvoke to 2.2.0_1 breaks running `inv` if `python@3.12` is not installed

### ISSUE THREAD

**Issue by bradleyneumaier** [Homebrew/homebrew-core#151248]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output
(seems like a different issue than https://github.com/Homebrew/homebrew-core/issues/151232)

`brew gist-logs pyinvoke`
```shell
Error: No logs.
```

`brew config`

```
HOMEBREW_VERSION: 4.1.16
ORIGIN: https://github.com/Homebrew/brew
HEAD: c0c8a4dd4dd19ec16deb429abd50cdf82ea33b27
Last commit: 2 days ago
Core tap JSON: 16 Oct 16:56 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_FORCE_BREWED_CURL: set
HOMEBREW_FORCE_BREWED_GIT: set
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_NO_INSECURE_REDIRECT: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.42.0 => /opt/homebrew/opt/git/bin/git
Curl: 8.4.0 => /opt/homebrew/opt/curl/bin/curl
macOS: 14.0-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
Rosetta 2: false
```

`brew doctor`

```
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

We have a command called `fixit` for `pyinvoke`. I tried running `inv fixit` after upgrading `pyinvoke`. This issue is applicable for any command for `pyinvoke`.

### What happened (include all command output)?

```
==> Upgrading pyinvoke
  2.2.0 -> 2.2.0_1

==> Pouring pyinvoke--2.2.0_1.arm64_sonoma.bottle.tar.gz
🍺  /opt/homebrew/Cellar/pyinvoke/2.2.0_1: 189 files, 1.6MB
==> Running `brew cleanup pyinvoke`...
Removing: /opt/homebrew/Cellar/pyinvoke/2.2.0... (872 files, 11MB)
```

```
inv fixit
exec: Failed to execute process '/opt/homebrew/bin/inv': The file specified the interpreter '/opt/homebrew/opt/python@3.12/bin/python3.12', which is not an executable command.
```

### What did you expect to happen?

I expected it to succeed without needing to install `python@3.12` manually first  (which causes it to work)

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew cleanup
brew autoremove
brew update
brew upgrade pyinvoke
inv fixit
```

**Comment by cho-m** [comment]
For formulae that support multiple Python bindings, this is a known limitation in Homebrew as there is an audit check that prevents adding a runtime dependency. If you make `python@3.12` a runtime dependency, it will error with:
```console
❯ brew audit --strict pyinvoke
pyinvoke
  * line 23, col 14: References to `python@3.10` should match the specified python dependency (`python@3.12`)
  * line 24, col 14: References to `python@3.11` should match the specified python dependency (`python@3.12`)
Error: 2 problems in 1 formula detected.
```

This issue needs someone familiar with RuboCop to first fix audit to allow a runtime dependency at https://github.com/Homebrew/brew/blob/master/Library/Homebrew/rubocops/lines.rb#L385-L397

Only other alternative I can think of right now is to only support single Python version. Usually, we add multiple Pythons to help migration effort (e.g. `python@3.11 --> python@3.12`) as Homebrew tries to use latest versions of formulae. Multiple versions were added to `pyinvoke` in https://github.com/Homebrew/homebrew-core/commit/b3a49f013b8166fa90f6565dc7cf0b69472d3571. We could migrate all dependents to `python@3.12` and revert this to only single Python.

**Comment by cho-m** [comment]
Should be fixed with #152286
